### ISSUE TITLE
Update to 2.3.148 breaks on macOS

### ISSUE THREAD

**Issue by rcarmo** [Azure/azure-functions-core-tools#939]
I just tried updating the CLI via `npm` (I have npm set to install globally on `$HOME/.npm-packages` since I will not allow it to touch system paths) and now the functions host is broken:

```
$ npm install -g azure-functions-core-tools
/Users/rcarmo/.npm-packages/bin/func -> /Users/rcarmo/.npm-packages/lib/node_modules/azure-functions-core-tools/lib/main.js
/Users/rcarmo/.npm-packages/bin/azfun -> /Users/rcarmo/.npm-packages/lib/node_modules/azure-functions-core-tools/lib/main.js
/Users/rcarmo/.npm-packages/bin/azurefunctions -> /Users/rcarmo/.npm-packages/lib/node_modules/azure-functions-core-tools/lib/main.js

> azure-functions-core-tools@2.3.148 postinstall /Users/rcarmo/.npm-packages/lib/node_modules/azure-functions-core-tools
> node lib/install.js

attempting to GET "https://functionscdn.azureedge.net/public/2.3.148/Azure.Functions.Cli.osx-x64.2.3.148.zip"
[==================] Downloading Azure Functions Cli
+ azure-functions-core-tools@2.3.148
updated 2 packages in 15.859s

$ func host start
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: spawn /Users/rcarmo/.npm-packages/lib/node_modules/azure-functions-core-tools/bin/func EACCES
    at Process.ChildProcess._handle.onexit (internal/child_process.js:229:19)
    at onErrorNT (internal/child_process.js:406:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
    at startup (internal/bootstrap/node.js:266:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:596:3)
Emitted 'error' event at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:235:12)
    at onErrorNT (internal/child_process.js:406:16)
    [... lines matching original stack trace ...]
    at bootstrapNodeJSCore (internal/bootstrap/node.js:596:3)
```

**Comment by Bomret** [comment]
Same here, same stack

**Comment by ankitkumarr** [comment]
I wasn't able to reproduce. But, could you try installing like this-
`npm i -g azure-functions-core-tools --unsafe-perm true`

**Comment by Bomret** [comment]
We have the tools installed project local with `npm i -D azure-functions-core-tools`. Adding `--unsafe-perm true` does not change anything. It seems that file ownership or permissions are changed so it cannot access something anymore or the binary references something new that is not allowed? I'm using the zsh shell. This problem definitely is new with the 2.3, never seen it before. Problem is that it also affected the build agent on the CI server. It wasn't able to build anymore because it also got the EACCESS error

**Comment by ankitkumarr** [comment]
@Bomret, I believe you are correct. There's something wrong with permissions. I did reproduce this issue and than added execute permissions to the `func` executable which seemed to work.
So for the trace above, `sudo chmod a+x /Users/rcarmo/.npm-packages/lib/node_modules/azure-functions-core-tools/bin/func` should mitigate the problem.

I don't think anything particular should have changed in this version. @ahmelsayed / @soninaren, is it something that we missed adding permissions in our build project?

Thanks for reporting!

**Comment by ankitkumarr** [comment]
Upon further review, I think I found the issue. Our install script holds a [dependency ](https://github.com/Azure/azure-functions-core-tools/blob/master/src/Azure.Functions.Cli/npm/package.json#L36)on an npm package `unzipper` which released a new version (`0.9.6`) yesterday. Looks like in their new version, `.on('close')` event is not firing which we use [here](https://github.com/Azure/azure-functions-core-tools/blob/master/src/Azure.Functions.Cli/npm/lib/install.js#L66-L70). 

I believe we should not be depending on third-party dependencies dynamically. We should either use `"unzipper": "0.9.2"` over `"unzipper": "^0.9.2"` and should create a lock file like `npm-shrinkwrap.json`. We will make the update. For the time being, please manually add execute permissions to the `/bin/func` file (elaborated in the previous comment). I apologize for the inconvenience and thank you for the report.

\cc @ahmelsayed

**Comment by Bomret** [comment]
@ankitkumarr thanks! Good to hear that this will be fixed soon.
